src/hotspot/share/prims/jvmtiEnvBase.cpp

```diff
@@ -659,33 +661,43 @@ JvmtiEnvBase::check_and_skip_hidden_frames(bool is_in_VTMS_transition, javaVFram
     return jvf;  // No frames to skip.
   }
   // Find jvf with a method annotated with @JvmtiMountTransition.
+  // Two cases with the annotated methods to skip on the top:
+  //  - is_in_VTMS_transition == false and two top annotated methods are yield and yield0
+  //  - is_in_VTMS_transition = true and some methods followed by a motifyJvmti* method
   for ( ; jvf != nullptr; jvf = jvf->java_sender()) {
-    if (jvf->method()->jvmti_mount_transition()) {  // Cannot actually appear in an unmounted continuation; they're never frozen.
+    if (jvf->method()->jvmti_mount_transition()) {
+      // The yield and yield0 may appear in an unmounted continuation.
       jvf = jvf->java_sender();  // Skip annotated method.
-      break;
+      continue;
     }
     if (jvf->method()->changes_current_thread()) {
       break;
     }
-    // Skip frame above annotated method.
+    if (is_in_VTMS_transition) {
+      // Skip frames above annotated method.
+    } else {
+      // Stop at first frame with non-annotated method.
+      break;
+    }
   }
   return jvf;
 }
 
 javaVFrame*
 JvmtiEnvBase::check_and_skip_hidden_frames(JavaThread* jt, javaVFrame* jvf) {
-  jvf = check_and_skip_hidden_frames(jt->is_in_VTMS_transition(), jvf);
+  bool is_virtual = java_lang_VirtualThread::is_instance(jt->jvmti_vthread());
+
+  if (is_virtual || jt->is_in_VTMS_transition()) { // filter out pure continuations
+    jvf = check_and_skip_hidden_frames(jt->is_in_VTMS_transition(), jvf);
```

Comment:
Wouldn't be easier to split method `check_and_skip_hidden_frames` to skip_yeilds and skip_transition frames? 
BTW Also it is unclear shouldn't the `@Hidden` methods be skipped from all jvmti frame streams?

Reply:
Good suggestion, thanks. I was also thinking about it but was not sure it can be simplified.
JVMTI does not support @Hidden annotation which is used in Java `printStackTrace()` api implementation.